### PR TITLE
Add facial recogition support for Watson

### PR DESCRIPTION
**Pull request by akb**

### CODE REVIEW

File: visagoapi/watson/watson.go [hunk 1]
```diff
@@ -41,104 +43,132 @@ func (p *Plugin) Perform(c *visagoapi.PluginConfig) (string, visagoapi.PluginRes
 		return "", nil, fmt.Errorf("must supply files/URLs")
 	}
 
-	client := &http.Client{}
+	requestID := nuid.Next()
```
Comment (akb) [inline visagoapi/watson/watson.go]: This is looking like git might have messed something up. It looks like the code was changed to re-use the http.Client (and some other cleanup) underneath me. Will fix.

File: visagoapi/googlevision/googlevision.go [hunk 2]
```diff
@@ -172,8 +172,8 @@ func (p *Plugin) Faces(requestID string) (faces map[string][]*visagoapi.PluginFa
 			poly := &visagoapi.BoundingPoly{}
 			for _, v := range faceA.BoundingPoly.Vertices {
 				vertex := visagoapi.Vertex{
-					X: v.X,
-					Y: v.Y,
+					X: int(v.X), // FIXME: this will break if images larger than 4.6 Exapixels are used
```
Comment (zquestz) [inline visagoapi/googlevision/googlevision.go]: NIT: would reword to: "FIXME: Add support for images larger than 4.6 Exapixels"

Comment (akb) [inline visagoapi/googlevision/googlevision.go]: haha, can do. it was kinda meant as a joke because that is absurdly large (4 billion x 4billion) and we'll never add it, it just justifies what might seem like a bad cast

File: visagoapi/result.go [hunk 3]
```diff
@@ -148,3 +155,22 @@ func mergeAssets(assets []*Asset) []*Asset {
 
 	return mergedAssets
 }
+
+func mergeFaces(faces []*PluginFaceResult, face *PluginFaceResult) []*PluginFaceResult {
+	if len(faces) == 0 {
+		return []*PluginFaceResult{face}
```
Comment (zquestz) [inline visagoapi/result.go]: Shouldn't this be `return []*PluginFaceResult{}`

Comment (zquestz) [inline visagoapi/result.go]: NM. I see what you are doing.

File: visagoapi/googlevision/googlevision.go (hunk 2 above)
Comment (zquestz) [inline visagoapi/googlevision/googlevision.go]: I know, it is just hard to read the joke when it scrolls off your screen. =)
